libs/angular/src/services/injection-tokens.ts

```diff
@@ -7,26 +7,39 @@ import {
 } from "@bitwarden/common/platform/abstractions/storage.service";
 import { StateFactory } from "@bitwarden/common/platform/factories/state-factory";
 
-export const WINDOW = new InjectionToken<Window>("WINDOW");
-export const OBSERVABLE_MEMORY_STORAGE = new InjectionToken<
+/**
+ * A (more) typesafe version of InjectionToken.
+ * The default angular implementation does not use the generic type to define the structure of the object,
+ * so the structural type system will not object to a mismatch in generic types.
+ * e.g. const test: InjectionToken<string> = new InjectionToken<number>(); // compiles just fine
+ * This is solved by assigning T to an arbitrary property.
+ */
+export class SafeInjectionToken<T> extends InjectionToken<T> {
+  readonly bitTypeRecord: T;
+}
```

Comment:
this concept is sometimes called a marker or tag, we use it a few times in the repository.

It's a little cleaner to assign to a unique symbol rather than a weird readonly that people could find.

```suggestion
declare const tag: unique symbol;
export class SafeInjectionToken<T> extends InjectionToken<T> {
  private readonly [tag]: T;
}
```

Reply:
Exactly what I wanted, ty